app/plugin_api_test.go

```diff
@@ -51,12 +58,13 @@ func setupMultiPluginApiTest(t *testing.T, pluginCodes []string, pluginManifests
 		require.True(t, activated)
 	}
 
-	app.SetPluginsEnvironment(env)
-
-	return pluginDir
+	return func() {
+		defer os.RemoveAll(pluginDir)
```

Comment:
Note too that the `defer` here is redundant -- it really just reverses the order of these two function calls.